bin/StringsScript/Tests/StringsScriptTests/StringsScriptTests.swift

```diff
@@ -0,0 +1,87 @@
+import XCTest
+@testable import StringsScriptCore
+import class Foundation.Bundle
+
+// swiftlint:disable line_length
+final class StringsScriptTests: XCTestCase {
+
+  var subject: Strings?
+
+  override func setUp() {
+    super.setUp()
+    self.subject = Strings()
+  }
+
+  func testStringsScript_run_throwsErrorWithInsufficientArguments() {
+    let stringsScript = StringsScript(arguments: [String]())
+
+    do {
+      try stringsScript.run()
+
+      XCTFail("Script should fail with insufficient number of arguments")
+    } catch {
+      XCTAssertEqual(StringsScriptError.insufficientArguments.localizedDescription, error.localizedDescription)
+    }
+  }
+
+  func testStringsFileContents() {
+    let strings = ["Add_attachments": "Add attachments…"]
+    let content = self.subject?.stringsFileContents(strings)
+    XCTAssertEqual(content, "\"Add_attachments\" = \"Add attachments…\";")
+  }
+
+  func testLocalePathsAndContents() {
+
+    let stringsByLocale = ["fr": ["Kickstarter_is_not_a_store": "Kickstarter n\'est pas un magasin."]]
+    let strings = Strings()
+    let localePath = "Kickstarter-iOS/Locales"
+
+    if let (locale, content) = strings.localePathsAndContents(with: localePath,
+                                                              stringsByLocale: stringsByLocale).first {
+      XCTAssertEqual(locale, "Kickstarter-iOS/Locales/fr.lproj/Localizable.strings")
+      XCTAssertEqual(content, "\"Kickstarter_is_not_a_store\" = \"Kickstarter n\'est pas un magasin.\";")
+    } else {
+      XCTFail("Locale and content cannot be nil")
+    }
+  }
+
+  func testDeserialize() {
+    //swiftlint:disable:next todo
+    //TODO
+  }
+
+  func testStaticStringsFileContents() {
+    let stringsByLocale = ["Base": ["Are_you_sure_you_wish_to_remove_this_card": "Are you sure you wish to remove this card from your payment method options?"]]
```

Comment:
Should we include a non-Base locale test in the generation here too?

Reply:
The `staticStringsFileContents` generates exclusively strings for the Base locale. If we test the function using a non-Base locale, it will simply generate the header comments and an empty `Strings` enum. Do you think it's worth to test it regardless?

Reply:
Oh, I see this line in that function: `try stringsByLocale["Base"]?.keys`. But the same function is used to generate the other locales too. I would want to test this function with a more complete dictionary. Maybe what we really want to do is add some dependency injection to `fetchStringsByLocale()` and pass in some stubbed JSON to test with?